cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -1007,6 +1010,38 @@ func deleteAllPodsBoundToNode(ctx *controllerContext, nodeName string) error {
 	return utilerrors.NewAggregate(errs)
 }
 
+func markFailedAndDeletePodWithCondition(ctx *controllerContext, pod *v1.Pod) error {
+	const fieldManager = "GCPControllerManager"
+	condition := corev1apply.PodCondition().
```

Comment:
nit: move the condition declaration down, or even directly pass into `WithConditions`.

Reply:
Done!